enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/RelationshipGroupRecordFormat.java

```diff
@@ -104,8 +106,10 @@ protected void doReadInternal( RelationshipGroupRecord record, PageCursor cursor
         }
         else
         {
+            int typeLowWord = cursor.getShort() & 0xFFFF;
+            int type = (((cursor.getByte() & 0xFF) << Short.SIZE) | typeLowWord);
```

Comment:
I usually find it safer to first read the `cursor.getByte() & 0xFF` into an `int` variable so that it's guaranteed that `<< Short.SIZE` won't try to left-shift a byte (which obviously means loss of bits). But perhaps just doing the `& 0xFF` converts it to an `int` always. I just know how sensitive these things are sometimes.